todoapp/codelab/25mar2017/03_activity_detail.md

```diff
@@ -0,0 +1,156 @@
+
+<img src="http://i.imgur.com/7GdKSgQ.png" height = 500px>
+
+Para verificar que efectivamente el usuario decidió dar permiso a la aplicación utilizaremos el método `onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults)` que trabaja similar a onActivityResult y verificara que tengamos permiso para utilizar location después de desaparecer el dialogo del permiso, de no ser así desactivara el switch igual que onActivityResult ya que las condiciones no se cumplen para poder utilizar el servicio de location. 
+```java
+public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        if (requestCode == REQUEST_LOCATION_PERMISSION && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+            getLocationFromService();
+        } else {
+            locationSwitch.setChecked(false);
+        }
+    }
+```
+
+En caso contrario, es decir, que el usuario de permiso volveremos a ejecutará `getLocationFromService()` que esta vez con permiso del usuario obtendremos el location y lo guardaremos en nuestro ToDo con el siguiente metodo `saveLocationInToDo(location)` que definiremos a continuación:
+```java
+private void saveLocationInToDo(Location location) {
+        if (location != null) {
+            todoItem.setLat(location.getLatitude());
+            todoItem.setLng(location.getLongitude());
+        }
+    }
+```
+
+Por ultimo nos devolvemos a el listener de nuestro locationSwitch y le proporcionamos la funcionalidad que construimos. Cuando el switch sea activado verificara si tenemos la configuración de Location habilitada, si es así obtendrá el location del servicio que tenemos en locationManager. Si no tenemos location activado mostrara el dialogo para preguntar si queremos activarlo en los ajustes de nuestro dispositivo. Si el switch es desactivado asignaremos valores de **0.0d**(sin location) a nuestro ToDo. Nuestro listener queda así:
+```java
+ locationSwitch.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton compoundButton, boolean isChecked) {
+                if (isChecked) {
+                    if (locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)) {
+                        getLocationFromService();
+                    } else {
+                        showLocationNeededDialog();
+                    }
+                } else {
+                    todoItem.setLat(0.0d);
+                    todoItem.setLng(0.0d);
+                }
+            }
+        });
+```
+Con esto terminamos la funcionalidad para guardar location dentro de nuestro ToDo.
```

Comment:
There is a lot of code in this 'page', can we at least give some links to the slides?